pkg/occlient/templates.go

```diff
@@ -139,6 +148,35 @@ func generateSupervisordDeploymentConfig(commonObjectMeta metav1.ObjectMeta, com
 	return dc
 }
 
+func updateSupervisorDeploymentConfig(updateParams SupervisorDUpdateParams) appsv1.DeploymentConfig {
+
+	dc := *updateParams.existingDc
+
+	dc.ObjectMeta = updateParams.commonObjectMeta
+
+	if len(dc.Spec.Template.Spec.Containers) > 0 {
+		dc.Spec.Template.Spec.Containers[0].Name = updateParams.commonObjectMeta.Name
+		dc.Spec.Template.Spec.Containers[0].Ports = updateParams.commonImageMeta.Ports
+		dc.Spec.Template.Spec.Containers[0].Env = updateParams.envVar
+		dc.Spec.Template.Spec.Containers[0].EnvFrom = updateParams.envFrom
+	}
+
+	containerIndex := -1
+	if updateParams.resourceRequirements != nil {
+		for index, container := range dc.Spec.Template.Spec.Containers {
+			if container.Name == updateParams.commonObjectMeta.Name {
+				containerIndex = index
+				break
+			}
+		}
+		if containerIndex != -1 {
+			dc.Spec.Template.Spec.Containers[containerIndex].Resources = *updateParams.resourceRequirements
+		}
+	}
+
+	return dc
+}
+
```

Comment:
Can you please add comments to the above function of what's happening what why it's needed?